src/infra/core/src/engine/engine_io_strategy.rs

```diff
@@ -0,0 +1,235 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::path::{Path, PathBuf};
+use std::sync::Arc;
+
+use container_runtime::*;
+use opendatafabric::*;
+
+use crate::domain::*;
+
+///////////////////////////////////////////////////////////////////////////////
+
+/// This IO strategy materializes all inputs as local file system files and pass
+/// them to the engines via mounted volumes.
+pub struct EngineIoStrategyLocalVolume {
+    dataset_repo: Arc<dyn DatasetRepository>,
+}
+
+impl EngineIoStrategyLocalVolume {
+    pub fn new(dataset_repo: Arc<dyn DatasetRepository>) -> Self {
+        Self { dataset_repo }
+    }
+
+    async fn materialize_object(
+        &self,
+        repo: &dyn ObjectRepository,
+        hash: &Multihash,
+        container_in_dir: &Path,
+        volumes: &mut Vec<VolumeSpec>,
+    ) -> Result<PathBuf, InternalError> {
+        let url = repo.get_internal_url(hash).await;
+        let host_path = kamu_data_utils::data::local_url::into_local_path(url).int_err()?;
```

Comment:
Should we use `CachedObject` here maybe?

Reply:
Yep, this will likely be a part of my next PR